src/Network/Transport/InMemory.hs

```diff
@@ -62,17 +67,26 @@ data LocalConnectionState
   | LocalConnectionClosed
   | LocalConnectionFailed
 
+newtype TransportInternals = TransportInternals (TVar TransportState)
+
 -- | Create a new Transport.
 --
 -- Only a single transport should be created per Haskell process
 -- (threads can, and should, create their own endpoints though).
 createTransport :: IO Transport
-createTransport = do
+createTransport = fst <$> createTransportExposeInternals
+
+-- | Create a new Transport exposing internal state.
+--
+-- This state can be used for with internal or debug functions, e.g.
```

Comment:
"Useful for testing and/or debugging purposes. Should not be used in production. No guarantee as to the stability of the internals API."